doc/source/whatsnew/v1.5.0.rst

```diff
@@ -996,6 +996,8 @@ Groupby/resample/rolling
 - Bug in :meth:`.DataFrameGroupBy.describe` and :meth:`.SeriesGroupBy.describe` produces inconsistent results for empty datasets (:issue:`41575`)
 - Bug in :meth:`DataFrame.resample` reduction methods when used with ``on`` would attempt to aggregate the provided column (:issue:`47079`)
 - Bug in :meth:`DataFrame.groupby` and :meth:`Series.groupby` would not respect ``dropna=False`` when the input DataFrame/Series had a NaN values in a :class:`MultiIndex` (:issue:`46783`)
+- Bug in :meth:`DataFrameGroupBy.resample` raises ``KeyError`` when getting the result from a key list which misses the resample key (:issue:`47362`)
```

Comment:
If we change the description to `a single element list`, it's not accurate since `resample` also raises for keys like `["val1", "val2"]`. It actually depends on whether the sample key ("date") is in the list, which means it will be fine if we pass `["val1", "val2", "date"]`.

Reply:
Ah okay. Maybe mentioning `with a list of keys` is better then